script/microsoft_tts.py

```diff
@@ -0,0 +1,29 @@
+"""Helper script to update supported languages for Microsoft Text-to-Speech (TTS)."""
+from pathlib import Path
+
+from lxml import html
+import requests
+
+from .hassfest.serializer import format_python_namespace
+
+URL = "https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/language-support"
+XPATH_QUERY = "//*[@data-tab='tts']//tr/td[1]/code/text()"
+
+req = requests.get(URL)
+req.raise_for_status()
+tree = html.fromstring(req.content)
+supported_languages_raw = tree.xpath(XPATH_QUERY)
+supported_languages = {s.lower() for s in supported_languages_raw}
+
+assert "en-us" in supported_languages
+assert "fa-ir" in supported_languages
+assert len(supported_languages) > 10
```

Comment:
It's a bit weird to sprinkle asserts here. If the point is to make sure fetching the language list was successful, add replace with clear prints and then exit the script with a failing result code.

If this is a leftover from developing, please remove the asserts.

Reply:
I added tests for Microsoft TTS, since there currently were none, adapted from the tests for Google Translate TTS. Moved these assertions to the tests.